src/atoms/Radio/__tests__/Radio.web.test.js

```diff
@@ -0,0 +1,358 @@
+          <Radio value="1">
+            <Radio.Option value="1" />
+          </Radio>,
+        );
+        const radio = getByRole('radio');
+        radio.focus();
+        expect(radio).toHaveFocus();
+        expect(container).toMatchSnapshot();
+      });
+    });
+  });
+
+  describe('helpText', () => {
+    it('should render radio with provided helpText', () => {
+      const helpText = 'This is some sample help text.';
+      const { container } = renderWithTheme(
+        <Radio value="1">
+          <Radio.Option value="1" helpText={helpText} />
+        </Radio>,
+      );
+      expect(container).toHaveTextContent(helpText);
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('errorText', () => {
+    it('should render radio with provided errorText', () => {
+      const errorText = 'This is some sample error text.';
+      const { container } = renderWithTheme(
+        <Radio value="1">
+          <Radio.Option value="1" errorText={errorText} />
+        </Radio>,
+      );
+      expect(container).toHaveTextContent(errorText);
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('onChange', () => {
+    describe('with defaultValue(uncontrolled)', () => {
+      it('should call onChange with updated check value', () => {
+        const onChange = jest.fn();
+        const { getByRole } = renderWithTheme(
+          <Radio onChange={onChange} defaultValue="1">
+            <Radio.Option value="2" />
+          </Radio>,
+        );
+        const radio = getByRole('radio');
+        expect(radio.checked).toBe(false);
+        fireEvent.click(radio);
+        expect(radio.checked).toBe(true);
+        expect(onChange).toHaveBeenCalledTimes(1);
+        expect(onChange).toHaveBeenCalledWith('2');
+      });
+
+      describe('with disabled', () => {
+        it('should not call onChange if radio is disabled', () => {
+          const onChange = jest.fn();
+          const { getByRole } = renderWithTheme(
+            <Radio onChange={onChange} defaultValue="1">
+              <Radio.Option value="2" disabled />
+            </Radio>,
+          );
+          const radio = getByRole('radio');
+          expect(radio.checked).toBe(false);
+          fireEvent.click(radio);
+          expect(radio.checked).toBe(false);
+          expect(onChange).not.toBeCalled();
+        });
+      });
+    });
+
+    describe('with value(controlled)', () => {
+      it('should call onChange with updated check value', () => {
+        const onChange = jest.fn();
+        const { getByRole } = renderWithTheme(
+          <Radio onChange={onChange} value="1">
+            <Radio.Option value="2" />
+          </Radio>,
+        );
+        const radio = getByRole('radio');
+        expect(radio.checked).toBe(false);
+        fireEvent.click(radio);
+        expect(radio.checked).toBe(false);
+        expect(onChange).toHaveBeenCalledTimes(1);
+        expect(onChange).toHaveBeenCalledWith('2');
```

Comment:
If `checked` is `false`, why would you expect `onChange` to get called with the new value?

Reply:
Since its controlled case, it does not manage its own state and calls `onChange` without setting its value first.